slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/dao/SignedAttestationsDao.java

```diff
@@ -12,22 +12,44 @@
  */
 package tech.pegasys.web3signer.slashingprotection.dao;
 
+import java.util.List;
 import java.util.Optional;
 import java.util.stream.Stream;
 
+import org.apache.tuweni.bytes.Bytes;
 import org.apache.tuweni.units.bigints.UInt64;
 import org.jdbi.v3.core.Handle;
 
 public class SignedAttestationsDao {
 
-  public Optional<SignedAttestation> findExistingAttestation(
-      final Handle handle, final int validatorId, final UInt64 targetEpoch) {
+  public List<SignedAttestation> findAttestationsForEpochWithDifferentSigningRoot(
```

Comment:
Similar to the BlocksDao looks like don't actually need all the entries so could do a LIMIT 1 and return Optional<SignedAttestation> using findFirst

Reply:
see above.